src/Paket.Core/Versioning/SemVer.fs

```diff
@@ -177,55 +244,59 @@ module SemVer =
                 for s in version.Split([|'.'|]) do
```

Comment:
Can you change the comment of this function.
I'm not sure if this PR changed this but the line

`///     parse "1.2.3-alpha002" > parse "1.2.3-alpha1"   // true`

is no longer correct.

Reply:
I'm conflicted about this :) the change is side effect of preferring SemVer2 ordering, but I really want to keep compatibility with SemVer1 workarounds -- not even M$ packages (netcore) use v2 format and rules in all cases, and there are many companies (incl. ours) having _thousands_ of improperly versioned packages in their repos. 
In this case, real-world semantics should win, I'd expect -- `alpha002` should be bigger than `alpha1`, as the zero-padded version was created as workaround of broken ordering  above `alpha9`, maybe after `alpha1` already existed. 
Luckily, the SemVer1/2 special case handler is one of the simpler parts of the code :)

Reply:
Yes I agree, but this is about PRERELEASE only. And SemVer2 has been out for some time now. You should not depend on PRERELEASEs too long you know :). Can you give examples of currently used packages where this particular instance is a problem?

Personally I would have no problem with releasing this and look where and if it hurts someone and if we cannot figure out a workaround we can still revert and look for a solution.

Reply:
I just verified that this PR did not change this behavior:
```
> oldParse "1.2.3-alpha002" > oldParse "1.2.3-alpha1";;
val it : bool = false
```

Therefore this change is already live and probably nothing to worry about.

Reply:
As much as I like there currently is a widely accepted semver2, it does not cover all of use cases, and for some companies, named "prereleases" are the majority of their packages, being produced & consumed by named feature/bugfix branches. 

Similar approach can be seen in bigger companies, where there is simply no feasible fast release cycle (netcore could stay _years_ in prerelease). 

From my PoV, removing the old workaround of parsing numbers off alphanumeric segments makes transition no newer scheme harder -- I'd like to safely assume that `1.2.3-beta001` is less-than `1.2.3-beta.2` :)